front/lib/swr/vaults.ts

```diff
@@ -96,38 +96,67 @@ export function useVaultInfo({
   };
 }
 
-export function useVaultDataSourceViews<
-  IncludeConnectorDetails extends boolean,
->({
+export function useVaultDataSourceViews({
   category,
   disabled,
-  includeConnectorDetails,
-  includeEditedBy,
   vaultId,
   workspaceId,
 }: {
   category?: Exclude<DataSourceViewCategory, "apps">;
   disabled?: boolean;
-  includeConnectorDetails?: IncludeConnectorDetails;
-  includeEditedBy?: boolean;
   vaultId: string;
   workspaceId: string;
 }) {
   const vaultsDataSourceViewsFetcher: Fetcher<
-    GetVaultDataSourceViewsResponseBody<IncludeConnectorDetails>
+    GetVaultDataSourceViewsResponseBody<false>
   > = fetcher;
 
   const queryParams = new URLSearchParams();
   if (category) {
     queryParams.set("category", category);
   }
 
-  if (includeConnectorDetails) {
-    queryParams.set("includeConnectorDetails", "true");
-  }
-  if (includeEditedBy) {
-    queryParams.set("includeEditedBy", "true");
-  }
+  const { data, error, mutate, mutateRegardlessOfQueryParams } =
+    useSWRWithDefaults(
+      `/api/w/${workspaceId}/vaults/${vaultId}/data_source_views?${queryParams.toString()}`,
+      vaultsDataSourceViewsFetcher,
+      { disabled }
+    );
+
+  const vaultDataSourceViews = useMemo(() => {
+    return (data?.dataSourceViews ??
+      []) as GetVaultDataSourceViewsResponseBody<false>["dataSourceViews"];
+  }, [data]);
+
+  return {
+    vaultDataSourceViews,
+    mutate,
+    mutateRegardlessOfQueryParams,
+    isVaultDataSourceViewsLoading: !error && !data,
+    isVaultDataSourceViewsError: error,
+  };
+}
+
+export function useVaultDataSourceViewsWithDetails({
```

Comment:
What was wrong with choosing what to include ? Feel like don't necessarily need another hook in this case..

Reply:
TLDR: It's to ensure type checking.

Long version:
`useVaultDataSourceViews()` with options returned different types based on the option, however, it's impossible to check type at runtime so it didn't really work in the current implementation.

Also, keeping opt-in options (editedBy, connectorDetails etc..) means the type returned was getting quite complex and we had to make everything optional.

After trying to make it works and considering how complicated it was getting, I decided that it was more simple to make a dedicated hook that returns a dedicated type (I checked usage).

I would even argue that `editedByUser` should be removed as an optional DataSourceViewType property and returned the same way as `usage` and the connector details. (it's only used with one hook)

Try to follow all the added code to supply editedByUser to DataSourceView for just one endpoint and yet not even have a non optional property :)

Reply:
Got me convinced 🤓